scripts/executor.py

```diff
@@ -167,39 +94,39 @@ def a(event):
     while True:
         try:
             if len(sched0):
-                fut = submit_next(sched0, pool)    # when time comes, fire and forget
+                fut = schedule.submit_next(sched0, pool)    # when time comes, fire and forget
                 if fut is not None:
                     futures.append(fut)
                     logger.info(f"Submitted one. {len(futures)} futures")
                 if len(sched0):
                     if sched0.iloc[0].name != nextmjd:
                         nextmjd = sched0.iloc[0].name
-                        logger.info(f"Next submission at MJD {nextmjd}, in {(nextmjd-Time.now().mjd)*24*3600}s")
+                        logger.info(f"Next session at MJD {nextmjd}, in {(nextmjd-Time.now().mjd)*24*3600}s")
                 else:
-                    logger.info("Schedule contains 0 commands.")
+                    logger.info("Schedule contains 0 session commands.")
 
             # clean up futures
-#            for fut in futures:
-#                if fut.done():
-#                    print(f"Completed command: {fut.result()}")
-#                elif fut.cancelled():
-#                    print(f"Cancelled command: {fut.result()}")
-            futures = [fut for fut in futures if not fut.done() or not fut.cancelled()]
+            for fut in futures:
+                if fut.ready():
+                    logger.info(f"Completed command: {fut.get(timeout=1)}")
+                    futures.remove(fut)
             sleep(0.49)  # at least two per second
         except KeyboardInterrupt:
             logger.info("Interrupting execution of schedule. Clearing schedule and waiting on submissions (Ctrl-C again to interrupt)...")
             schedule.put_sched(DataFrame([]))
             for wid in ls.watch_ids:
                 ls.cancel(wid)
             try:
-                for fut in as_completed(futures):
-                    logger.info(f"Completed command: {fut.result()}")
+                for fut in futures:
+                    logger.info(f"Completed command: {fut.get(timeout=1)}")
             except KeyboardInterrupt:
                 logger.info(f"Interrupting again. Cancelling {len(futures)} submissions...")
-                for fut in futures:
-                    res = fut.cancel()
-                    if not res:
-                        logger.warning("\tCould not cancel a submission...")
+# not available in spawned processes?
```

Comment:
multiprocessing.pool.apply_aynsc doesn return Future, it returns https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.AsyncResult.